File: go/reverse-string/reverse_string.go
// Package reverse implements the Reverse String exercise.
package reverse

// Reverse a string.
func Reverse(input string) string {
	runes := []rune(input)
	length := len(runes)
	for i := 0; i < length/2; i++ {
		runes[i], runes[length-1-i] = runes[length-1-i], runes[i]
	}

	return string(runes)
}

File: clojure/reverse-string/src/reverse_string.clj
(ns reverse-string
  "The Reverse String exercise from Exercism")

(defn reverse-string
  "Return a string with the characters in a given string in reverse order."
  [s]
  (loop [chars s, reversed '()]
    (if-let [[c & cs] chars]
      (recur cs (cons c reversed))
      (apply str reversed))))
